client/state/plugins/premium/actions.js

```diff
@@ -209,27 +209,59 @@ function configure( site, plugin, dispatch ) {
 			action: 'register',
 		} );
 	}
-	site.setOption( { option_name: option, option_value: optionValue }, ( error, data ) => {
-		if ( ( 'vaultpress' === plugin.slug ) && versionCompare( plugin.version, '1.8.3', '>' ) ) {
-			const response = JSON.parse( data.option_value );
-			if ( 'response' === response.action && 'broken' === response.status ) {
-				error = new Error( response.error );
-				error.name = 'RegisterError';
+
+	const saveOption = () => {
```

Comment:
Unless I'm misreading this, it looks like we can define this outside of this function with: 
```
saveOption( site, plugin, dispatch ) {
    return return site.setOption( { option_name: option, option_value: optionValue }, ( error, data ) => { ... } );
);
}
```